### PR TITLE
Prototype for dnf support

### PR DESCRIPTION
**Pull request by DamienCassou**
2018-09-19-115653_740x539_scrot
2018-09-19-115729_4480x1080_scrot

### CODE REVIEW

**Comment by DamienCassou** [comment]
This is early stage. No action is available, only listing. Can you please give me a minimal list of things I have to implement before you can merge this PR?

File: helm-system-packages-dnf.el [hunk 1]
```diff
@@ -0,0 +1,77 @@
+;;; helm-system-packages-dnf.el --- Helm UI for Arch Linux' dnf. -*- lexical-binding: t -*-
```
Comment (Ambrevar) [inline helm-system-packages-dnf.el]: Not Arch Linux, or is it?

File: helm-system-packages-dnf.el [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+;; Copyright (C) 2012 ~ 2014 Thierry Volpiatto <thierry.volpiatto@gmail.com>
+;;               2017 ~ 2018 Pierre Neidhardt <mail@ambrevar.xyz>
+
+;; Author: Pierre Neidhardt <mail@ambrevar.xyz>
```
Comment (Ambrevar) [inline helm-system-packages-dnf.el]: Replace the copyright holders and the author :)

File: helm-system-packages-dnf.el [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;; Helm UI for Arch Linux' dnf.
```
Comment (Ambrevar) [inline helm-system-packages-dnf.el]: Arch Linux?

**Review by Ambrevar — COMMENTED** [review]
Thank you for this contribution!
Looks like you are off to a good start.
I can't test it but it seems that you got the starting point up and running.

**Comment by Ambrevar** [comment]
Before I merge this, could you please implement the information display at least?  Let me know if you need help with that.

**Comment by Ambrevar** [comment]
By the way, your screenshot does not shot alignment, but your code seems to do it.  Is alignment working?  And there is no `:` separator either, right?

**Comment by Ambrevar** [comment]
Forgot one thing: Feel free to update the readme to mention the new `dnf` support!

**Comment by DamienCassou** [comment]
> Before I merge this, could you please implement the information display at least? Let me know if you need help with that.

done. Please see screenshot above.

> By the way, your screenshot does not shot alignment, but your code seems to do it. Is alignment working? And there is no : separator either, right?

I implemented alignment after I took the screenshot. I've updated the screenshot.

> Forgot one thing: Feel free to update the readme to mention the new dnf support!

done

File: helm-system-packages-dnf.el [hunk 4]
```diff
@@ -0,0 +1,135 @@
+;;; helm-system-packages-dnf.el --- Helm UI for RPM-based distros using dnf. -*- lexical-binding: t -*-
+
+;; Copyright (C) 2018 Damien Cassou <damien@cassou.me>
+
+;; Author: Damien Cassou <damien@cassou.me>
```
Comment (Ambrevar) [inline helm-system-packages-dnf.el:5]: Please add me as `Maintainer: Pierre Neidhardt <mail@ambrevar.xyz`.

File: helm-system-packages-dnf.el [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                      (save-match-data
+                        (re-search-forward "^$")
+                        (match-beginning 0))))))))
+
+(defun helm-system-packages-dnf-info (candidate)
+  "Print information about helm CANDIDATE.
+With prefix argument, insert the output at point.
+Otherwise display in `helm-system-packages-buffer'."
+  (helm-system-packages-show-information
+   `((whatever . ,(helm-system-packages-dnf--info
```
Comment (Ambrevar) [inline helm-system-packages-dnf.el]: What is `whatever`?  I think you want to use `uninstalled` for packages without categories.

File: helm-system-packages-dnf.el (hunk 4 above)
Comment (DamienCassou) [reply helm-system-packages-dnf.el:5]: fixed

File: helm-system-packages-dnf.el (hunk 5 above)
Comment (DamienCassou) [reply helm-system-packages-dnf.el]: The docstring of `helm-system-packages-show-information` contains

> DESC-ALIST's keys are ignored

I thought that if it is ignored, I can put whatever :-). Can you please fix the docstring?

Also, I'm wondering what is this `uninstalled`. These packages could be installed, I don't know.

**Comment by Ambrevar** [comment]
The screenshot seems to be the same as before :p

**Comment by Ambrevar** [comment]
Merged, thank you very much for your awesome contribution!

Keep going, at least `install` and `uninstall` would be very welcome :)

**Comment by DamienCassou** [comment]
> The screenshot seems to be the same as before :p

sorry, I forgot to save the change :-). Now it's good.

**Comment by Ambrevar** [comment]
Looks great, thanks!

**Pull request merged**